### PR TITLE
HOTFIX: Fix crash when API returns Mythic progressionIndex

### PR DESCRIPTION
**Pull request by laserian403**
This PR fixes a crash when Tacticus API sync returns a Mythic `progressionIndex`.

Previously, the app code would intentionally crash. Now, it absorbs the value and clamps the derived result to Legendary Blue Wings.

Also added unit tests for this, since I think increasing test coverage might help us slowly smoke out bugs in the system.

A small change to the Github CI config was also required, to allow checks to succeed on this hotfix branch without deploying.

## Summary by CodeRabbit

* Bug Fixes
  * Improved handling of out-of-range progression data to prevent errors and ensure consistent rarity display.
* Tests
  * Added comprehensive unit tests covering progression-to-rarity mapping and boundary conditions.
* Chores
  * Updated deployment workflow to skip deploys when required secrets are missing.

### CODE REVIEW

File: src/v2/features/tacticus-integration/tacticus-integration.service.ts [hunk 1]
```diff
@@ -10,21 +10,34 @@
...
                 rarity = i as Rarity;
                 break;
             }
         }
 
-        const rarityStars: RarityStars = (progressionIndex - rarity) as RarityStars;
+        // As above, we clamp the rarity stars to the rarest we support. While this is inaccurate, it allows
+        // the planner to absorb unsupported values in the period between their appearance in the Tacticus API,
+        // and the planner adding support.
+        const rarityStars: RarityStars = (maxSupportedIndex - rarity) as RarityStars;
```
[the remark this answers is not in the document]
Comment (laserian403) [reply src/v2/features/tacticus-integration/tacticus-integration.service.ts:41]: I like your suggestion of adding tests. Please generate these for me.

File: src/v2/features/tacticus-integration/tacticus-integration.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        it('should clamp higher indices to Legendary Blue Wings', () => {
+            // Indices 16-19 are being incrementally rolled out for Mythic, so we
+            // can expect them to show up in Tacticus API responses in future.
+            // These test expectations will need to be modified as we add support for them.
+            const highVals = [16, 17, 18, 19, 20, 999];
+            for (let i = 0; i < highVals.length; i++) {
+                const result = TacticusIntegrationService.convertProgressionIndex(16);
+
+                expect(result).toEqual([Rarity.Legendary, RarityStars.BlueStar]);
+            }
```
[the remark this answers is not in the document]
Comment (laserian403) [reply src/v2/features/tacticus-integration/tacticus-integration.service.spec.ts]: Good catch thanks. Fixed in https://github.com/svehera/tacticusplanner/pull/415/commits/0569647b33157b81a460f432cd5fdce4d7d4e2b9

**Review by unrstuart — APPROVED** [review]
